CorsixTH/Src/th_lua_gfx.cpp

```diff
@@ -781,6 +781,14 @@ static int l_surface_get_renderer_details(lua_State *L)
     return 1;
 }
 
+// Lua to THRenderTarget->setWindowGrab
+static int l_surface_set_capture_mouse(lua_State *L)
+{
+    THRenderTarget* pCanvas = luaT_testuserdata<THRenderTarget>(L);
+    pCanvas->setWindowGrab((lua_isnoneornil(L, 2) != 0) ? false : (lua_toboolean(L, 2) != 0));
+    return 1;
```

Comment:
Return the number of values you are pushing back to lua. In this case 0.

Reply:
I'm not up to scratch on lua/c integration so I took my lead from an existing routine (l_surface_set_blue_filter_active) which is another void method call with same argument type, not doing a push back to lua and returns 1. I see others that have a lua_settop(L,1) return 1 when there is no explicit push. Maybe that is another bug.

Reply:
Shouldn't be a bug, it will just pass the top of the stack back as the return value, which is useless.

Moving the top of the stack to 1 and returning 1 might be to return self, for a fluent interface.